projects/groprotocol/helpers.js

```diff
@@ -0,0 +1,236 @@
+            .times(BigNumber(_reserve0))
+            .div(BigNumber(supply));
+          sdk.util.sumSingleBalance(
+            balances,
+            await transformAddress(token0),
+            token0Balance.toFixed(0)
+          );
+        }
+        if (!excludeTokens.includes(token1)) {
+          const token1Balance = BigNumber(lpPosition.balance)
+            .times(BigNumber(_reserve1))
+            .div(BigNumber(supply));
+          sdk.util.sumSingleBalance(
+            balances,
+            await transformAddress(token1),
+            token1Balance.toFixed(0)
+          );
+        }
+      } catch (e) {
+        console.log(
+          `Failed to get data for LP token at ${lpPosition.token} on chain ${chain}`
+        );
+        throw e;
+      }
+    })
+  );
+}
+
+const crvPools = {
+  // PWRD-3CRV
+  "0xbcb91e689114b9cc865ad7871845c95241df4105": {
+    swapContract: "0x001C249c09090D79Dc350A286247479F08c7aaD7",
+    underlyingTokens: [
+      "0xf0a93d4994b3d98fb5e3a2f90dbc2d69073cb86b",
+      "0x6c3F90f043a72FA612cbac8115EE7e52BDe6E490",
+    ],
+  },
+};
+
+async function unwrapCrvLPs(
+  balances,
+  lpPositions,
+  block,
+  chain = "ethereum",
+  transformAddress = (addr) => addr,
+  excludeTokens = []
+) {
+  await Promise.all(
+    lpPositions.map(async (lp) => {
+      try {
+        await unwrapCrv(
+          balances,
+          lp.token,
+          lp.balance,
+          block,
+          chain,
+          transformAddress,
+          excludeTokens
+        );
+      } catch (e) {
+        console.log(
+          `Failed to get data for LP token at ${lp.token} on chain ${chain}`
+        );
+        throw e;
+      }
+    })
+  );
+}
+
+async function unwrapCrv(
+  balances,
+  crvToken,
+  balance3Crv,
+  block,
+  chain = "ethereum",
+  transformAddress = (addr) => addr,
+  excludeTokensRaw = []
+) {
+  const excludeTokens = excludeTokensRaw.map((t) => t.toLowerCase());
+  if (crvPools[crvToken.toLowerCase()] === undefined) {
+    return;
+  }
+  const underlyingTokens = crvPools[crvToken.toLowerCase()].underlyingTokens;
+  const crvTotalSupply = sdk.api.erc20.totalSupply({
+    target: crvToken,
+    block,
+    chain,
+  });
+  const underlyingSwapTokens = (
+    await sdk.api.abi.multiCall({
+      calls: underlyingTokens.map((token) => ({
+        target: token,
+        params: [crvToken],
+      })),
+      block,
+      chain,
+      abi: "erc20:balanceOf",
+    })
+  ).output;
+  const resolvedCrvTotalSupply = (await crvTotalSupply).output;
+  underlyingSwapTokens.forEach((call) => {
+    if (excludeTokens.includes(call.input.target.toLowerCase())) {
+      return;
+    }
+    const underlyingBalance = BigNumber(call.output || 0)
+      .times(balance3Crv)
+      .div(resolvedCrvTotalSupply);
+    sdk.util.sumSingleBalance(
+      balances,
+      transformAddress(call.input.target),
+      underlyingBalance.toFixed(0)
+    );
+  });
+}
+
+module.exports = {
```

Comment:
wouldn't it be better to use these functions from our helper scripts? are these changed?

Reply:
We had to make a number of changes. Happy to feed those back to see if they could be included in core helpers

Reply:
yep this helpers has some mayor changes from original ones so had to be inlined or ad-hoc for us.

Reply:
@0xngmi added some comments about why we need this ad-hoc helpers

Reply:
got it!